src/bbs2gh/Commands/MigrateRepoCommand.cs

```diff
@@ -208,10 +213,11 @@ public override MigrateRepoCommandHandler BuildHandler(MigrateRepoCommandArgs ar
         if (args.SshUser.HasValue() || args.SmbUser.HasValue())
         {
             var bbsArchiveDownloaderFactory = sp.GetRequiredService<BbsArchiveDownloaderFactory>();
-            var bbsHost = new Uri(args.BbsServerUrl).Host;
+            var sshHost = args.SshHost.HasValue() ? args.SshHost : new Uri(args.BbsServerUrl).Host;
+
             bbsArchiveDownloader = args.SshUser.HasValue()
-                ? bbsArchiveDownloaderFactory.CreateSshDownloader(bbsHost, args.SshUser, args.SshPrivateKey, args.SshPort, args.BbsSharedHome)
-                : bbsArchiveDownloaderFactory.CreateSmbDownloader(bbsHost, args.SmbUser, args.SmbPassword, args.SmbDomain, args.BbsSharedHome);
+                ? bbsArchiveDownloaderFactory.CreateSshDownloader(sshHost, args.SshUser, args.SshPrivateKey, args.SshPort, args.BbsSharedHome)
+                : bbsArchiveDownloaderFactory.CreateSmbDownloader(sshHost, args.SmbUser, args.SmbPassword, args.SmbDomain, args.BbsSharedHome);
```

Comment:
This is going to be very confusing since `ssh-host` is being used both for SMB and SSH. Either we need to drop the ssh to indicate it's for both SMB and SSM or create yet another one for SMB (e.g. `--smb-host`).

Reply:
Oops! I didn't mean to do that. Would it be okay with you if I just add the `--ssh-user` option for now, which only works for SSH, and then we can add a similar feature for SMB later?

Reply:
is it worth creating two args? can't we just create one arg that can be reused for both? here are a couple of examples:
- `--bbs-host`
- `--download-host`

Reply:
I agree with you that it would probably be nice to have a single argument that covers SMB and SSH - but I want to think about that a bit more and consider what name I like best. 

I'll come back to you tomorrow!

Reply:
it sounds great 🎉